ledger/complete/ledger.go

```diff
@@ -200,11 +200,6 @@ func (l *Ledger) Get(query *ledger.Query) (values []ledger.Value, err error) {
 func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, trieUpdate *ledger.TrieUpdate, err error) {
 	start := time.Now()
 
-	if update.Size() == 0 {
-		// return current state root unchanged
-		return update.State(), nil, nil
```

Comment:
This special case creates an additional case where the returned `trieUpdate` could be `nil`. 

By removing this special case, we always return a non-nil trieUpdate. If the update is empty, then the returned trie update would also be empty, meaning it contains an empty list of updated path and payloads.

Reply:
> will this impact performance?

Yes, removing this special case would affect performance.  

> We remove this special case by removing this check and handling, so that a non-nil trieUpdate is always returned.

If the purpose is to return a non-nil `trieUpdate`, we can create and return an empty `trieUpdate` instead.

Reply:
will this affect the `trieUpdate` returned in the service chunk? `ExecutionData` has special case handling for that. Setting it to an empty set is also fine, just want to make sure this isn't going to break verification

Reply:
Yeah, a `nil` trieUpdate is different from empty trieUpdate, so we will product different chunk execution data from  `CommitDelta` method call. 

If chunk execution data is different, then execution data id will be different which cause execution result to be different. 

So this change can only be deployed with spork or HCU